--- ./applestreamingclient/CMakeLists.txt.orig	2011-02-21 17:14:43.495096363 +0200
+++ ./applestreamingclient/CMakeLists.txt	2011-02-11 15:36:21.000000000 +0200
@@ -2,17 +2,18 @@
 PROJECT(applestreamingclient)
 
 #make the list with all the source files
-FILE(GLOB_RECURSE SOURCES "../../../../sources/applications/applestreamingclient/src/*.cpp" )
+FILE(GLOB_RECURSE SOURCES "src/*.cpp" )
 
 #prepare the include directories
 INCLUDE_DIRECTORIES(
-	../../../../3rdparty/lua-dev
-	../../../../sources/common/include
-	../../../../sources/thelib/include
-	../../../../sources/applications/applestreamingclient/include
+	../../lua/src
+	../../common/include
+	../../thelib/include
+	include
 	)
 
 #create THE library
 ADD_LIBRARY(applestreamingclient SHARED ${SOURCES})
 TARGET_LINK_LIBRARIES(applestreamingclient thelib)
 
+INSTALL(TARGETS applestreamingclient LIBRARY DESTINATION lib/rtmpd)
