--- ./admin/CMakeLists.txt.orig	2011-02-21 17:14:43.483094276 +0200
+++ ./admin/CMakeLists.txt	2011-02-11 15:36:21.000000000 +0200
@@ -2,16 +2,16 @@
 PROJECT(admin)
 
 #make the list with all the source files
-FILE(GLOB_RECURSE SOURCES "../../../../sources/applications/admin/src/*.cpp" )
+FILE(GLOB_RECURSE SOURCES "src/*.cpp" )

 SET(CMAKE_INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/${CRTMPSERVER_ROOT}/${PROJECT_NAME}")
 
 #prepare the include directories
 INCLUDE_DIRECTORIES(
-	../../../../3rdparty/lua-dev
-	../../../../sources/common/include
-	../../../../sources/thelib/include
-	../../../../sources/applications/admin/include
+	../../lua/src
+	../../common/include
+	../../thelib/include
+	include
 	)
 
 #create THE library
