--- CMakeLists.txt.orig	2011-02-24 19:21:10.530593832 +0200
+++ CMakeLists.txt	2011-02-24 19:20:16.614592846 +0200
@@ -2,16 +2,16 @@
 PROJECT(proxypublish)
 
 #make the list with all the source files
-FILE(GLOB_RECURSE SOURCES "../../../../sources/applications/proxypublish/src/*.cpp" )
+FILE(GLOB_RECURSE SOURCES "src/*.cpp" )
 
 SET(CMAKE_INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/${CRTMPSERVER_ROOT}/${PROJECT_NAME}")
 
 #prepare the include directories
 INCLUDE_DIRECTORIES(
-	../../../../3rdparty/lua-dev
-	../../../../sources/common/include
-	../../../../sources/thelib/include
-	../../../../sources/applications/proxypublish/include
+	../../lua/src
+	../../common/include
+	../../thelib/include
+	include
 	)
 
 #create THE library
