--- ./vptests/CMakeLists.txt.orig	2011-02-21 17:14:43.559092668 +0200
+++ ./vptests/CMakeLists.txt	2011-02-11 15:36:21.000000000 +0200
@@ -2,16 +2,16 @@
 PROJECT(vptests)
 
 #make the list with all the source files
-FILE(GLOB_RECURSE SOURCES "../../../../sources/applications/vptests/src/*.cpp" )
+FILE(GLOB_RECURSE SOURCES "src/*.cpp" )
 
 SET(CMAKE_INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/${CRTMPSERVER_ROOT}/${PROJECT_NAME}")
 
 #prepare the include directories
 INCLUDE_DIRECTORIES(
-	../../../../3rdparty/lua-dev
-	../../../../sources/common/include
-	../../../../sources/thelib/include
-	../../../../sources/applications/vptests/include
+	../../lua/src
+	../../common/include
+	../../thelib/include
+	include
 	)
 
 #create THE library
