--- ./samplefactory/CMakeLists.txt.orig	2011-02-21 17:14:43.547091560 +0200
+++ ./samplefactory/CMakeLists.txt	2011-02-11 15:36:21.000000000 +0200
@@ -2,18 +2,19 @@
 PROJECT(samplefactory)
 
 #make the list with all the source files
-FILE(GLOB_RECURSE SOURCES "../../../../sources/applications/samplefactory/src/*.cpp" )
+FILE(GLOB_RECURSE SOURCES "src/*.cpp" )
 
 #prepare the include directories
 INCLUDE_DIRECTORIES(
 	${OPENSSL_INCLUDE_PATH}
-	../../../../3rdparty/lua-dev
-	../../../../sources/common/include
-	../../../../sources/thelib/include
-	../../../../sources/applications/samplefactory/include
+	../../lua/src
+	../../common/include
+	../../thelib/include
+	include
 	)
 
 #create THE library
 ADD_LIBRARY(samplefactory SHARED ${SOURCES})
 TARGET_LINK_LIBRARIES(samplefactory thelib)
 
+INSTALL(TARGETS samplefactory LIBRARY DESTINATION lib/rtmpd)
