--- CMakeLists.txt.orig	2011-02-21 15:44:33.990593316 +0200
+++ CMakeLists.txt	2011-02-21 15:43:42.426592369 +0200
@@ -126,12 +126,12 @@
 #SET(CMAKE_VERBOSE_MAKEFILE TRUE)
 
 #generate defines.h
-IF (EXISTS ../../sources/common/include/defines.h)
+IF (EXISTS ${CMAKE_SOURCE_DIR}/common/include/defines.h)
 	MESSAGE(STATUS "Defines are in place")
-ELSE (EXISTS ../../sources/common/include/defines.h)
+ELSE (EXISTS ${CMAKE_SOURCE_DIR}/common/include/defines.h)
 	MESSAGE(STATUS "We need to rebuild the defines")
-	EXECUTE_PROCESS(COMMAND lua ../../constants/constants.lua cpp ../../sources/common/include/defines.h)
-ENDIF(EXISTS ../../sources/common/include/defines.h)
+	EXECUTE_PROCESS(COMMAND lua ${CMAKE_SOURCE_DIR}/constants/constants.lua cpp ${CMAKE_SOURCE_DIR}/common/include/defines.h)
+ENDIF(EXISTS ${CMAKE_SOURCE_DIR}/common/include/defines.h)
 
 
 #ADD_DEFINITIONS(-DHAS_SAFE_LOGGER)
@@ -143,7 +143,7 @@
 #ADD_SUBDIRECTORY(androidapplestreaming androidapplestreaming)
 
 ADD_DEFINITIONS(-DVM_LUA)
-ADD_SUBDIRECTORY(vm vm)
-ADD_SUBDIRECTORY(vmtests vmtests)
-ADD_SUBDIRECTORY(tests)
+#ADD_SUBDIRECTORY(vm vm)
+#ADD_SUBDIRECTORY(vmtests vmtests)
+#ADD_SUBDIRECTORY(tests)
 
