--- rtmpserver.lua.orig	2011-02-24 19:48:22.379095002 +0200
+++ rtmpserver.lua	2011-02-24 19:54:46.674594262 +0200
@@ -49,7 +49,7 @@
 			type="file",
 			level=6,
 			-- the file where the log messages are going to land
-			fileName="/tmp/crtmpserver.log"
+			fileName="/var/log/crtmpserver/crtmpserver.log"
 		}
 	},
 	
@@ -58,7 +58,7 @@
 	{
 		-- this is the root directory of all applications
 		-- usually this is relative to the binary execuable
-		rootDirectory="applications",
+		rootDirectory="/usr/lib/crtmpserver/applications",
 		
 		
 		--this is where the applications array starts
