--- CMakeLists.txt.orig	2011-02-24 19:47:48.706592792 +0200
+++ CMakeLists.txt	2011-02-24 19:47:25.522593404 +0200
@@ -3,14 +3,15 @@
 
 #prepare the include directories
 INCLUDE_DIRECTORIES(
-        ../../../3rdparty/lua-dev
-        ../../../sources/common/include
-	../../../sources/thelib/include
+	../lua/src
+	../common/include
+	../thelib/include
         )
 
-EXECUTE_PROCESS(COMMAND svnversion -n ../../sources OUTPUT_VARIABLE TEMP_FRAMEWORK_VER)
+EXECUTE_PROCESS(COMMAND svnversion -n ../thelib/src OUTPUT_VARIABLE TEMP_FRAMEWORK_VER)
 ADD_DEFINITIONS(-DRTMPSERVER_VERSION="0.${TEMP_FRAMEWORK_VER}")
-ADD_EXECUTABLE(rtmpserver ../../../sources/rtmpserver/src/rtmpserver.cpp)
+ADD_EXECUTABLE(rtmpserver src/rtmpserver.cpp)
 TARGET_LINK_LIBRARIES(rtmpserver thelib ${OPENSSL_LIBRARY_PATH} ${CRYPTO_LIBRARY_PATH})
-INSTALL(TARGETS rtmpserver RUNTIME DESTINATION bin)
+SET_TARGET_PROPERTIES(rtmpserver PROPERTIES OUTPUT_NAME crtmpserver)
+INSTALL(TARGETS rtmpserver RUNTIME DESTINATION sbin RENAME crtmpserver)
 
