--- CMakeLists.txt.orig	2011-02-21 16:54:55.263092378 +0200
+++ CMakeLists.txt	2011-02-21 17:00:03.094593804 +0200
@@ -2,7 +2,7 @@
 PROJECT(thelib)
 
 #make the list with all the source files
-FILE(GLOB_RECURSE SOURCES "../../../sources/thelib/src/*.cpp" )
+FILE(GLOB_RECURSE SOURCES "src/*.cpp" )
 
 ADD_DEFINITIONS(-DCreateRTCPPacket=CreateRTCPPacket_mystyle_only_once)
 ADD_DEFINITIONS(-DFeedDataAudioMPEG4Generic=FeedDataAudioMPEG4Generic_one_by_one)
@@ -12,13 +12,14 @@
 #prepare the include directories
 INCLUDE_DIRECTORIES(
 	${OPENSSL_INCLUDE_PATH}
-	../../../3rdparty/lua-dev
-	../../../sources/common/include
-	../../../sources/thelib/include
+	../lua/src
+	../common/include
+	include
 	)
 
 #create THE library
 ADD_LIBRARY(thelib SHARED ${SOURCES})
 TARGET_LINK_LIBRARIES(thelib common ${DL_LIBRARY_PATH} ${OPENSSL_LIBRARY_PATH} ${CRYPTO_LIBRARY_PATH})
-ADD_PRECOMPILED_HEADER(thelib ../../../sources/common/include/common.h)
+ADD_PRECOMPILED_HEADER(thelib ../common/include/common.h)
 
+INSTALL(TARGETS thelib LIBRARY DESTINATION lib)
