Description: disable vm, tests and vmtests.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -184,10 +184,10 @@
 #ADD_DEFINITIONS(-DHAS_SAFE_LOGGER)
 ADD_SUBDIRECTORY(common common)
 ADD_SUBDIRECTORY(thelib thelib)
-ADD_SUBDIRECTORY(vm vm)
+#ADD_SUBDIRECTORY(vm vm)
 ADD_SUBDIRECTORY(applications applications)
 ADD_SUBDIRECTORY(crtmpserver crtmpserver)
-ADD_SUBDIRECTORY(vmtests vmtests)
-ADD_SUBDIRECTORY(tests)
+#ADD_SUBDIRECTORY(vmtests vmtests)
+#ADD_SUBDIRECTORY(tests)
 #ADD_SUBDIRECTORY(androidapplestreaming androidapplestreaming)
 
