Description: disable vm, tests and vmtests.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -170,10 +170,10 @@
 ADD_SUBDIRECTORY(common common)
 ADD_SUBDIRECTORY(thelib thelib)
 ADD_SUBDIRECTORY(crtmpserver crtmpserver)
-ADD_SUBDIRECTORY(vm vm)
-ADD_SUBDIRECTORY(vmtests vmtests)
+#ADD_SUBDIRECTORY(vm vm)
+#ADD_SUBDIRECTORY(vmtests vmtests)
 ADD_SUBDIRECTORY(applications applications)
 #ADD_SUBDIRECTORY(androidapplestreaming androidapplestreaming)
 
-ADD_SUBDIRECTORY(tests)
+#ADD_SUBDIRECTORY(tests)
 
