Description: change log file default location.

--- a/crtmpserver/crtmpserver.lua
+++ b/crtmpserver/crtmpserver.lua
@@ -31,7 +31,7 @@
 			type="file",
 			level=6,
 			-- the file where the log messages are going to land
-			fileName="/tmp/crtmpserver",
+			fileName="/var/log/crtmpserver/crtmpserver.log",
 			--newLineCharacters="\r\n",
 			fileHistorySize=10,
 			fileLength=1024*256,
@@ -44,7 +44,7 @@
 	{
 		-- this is the root directory of all applications
 		-- usually this is relative to the binary execuable
-		rootDirectory="applications",
+		rootDirectory="/usr/lib/crtmpserver/applications",
 		
 		
 		--this is where the applications array starts
