Description: disable vmlua in configuration file

--- a/crtmpserver/crtmpserver.lua
+++ b/crtmpserver/crtmpserver.lua
@@ -353,7 +353,7 @@
 			--validateHandshake=true,
 			--default=true,
 		},
-		{
+		--[[{
 			name="vmapp",
 			description="An application demonstrating the use of virtual machines",
 			protocol="dynamiclinklibrary",
@@ -372,7 +372,7 @@
 					protocol="inboundTcpTs"
 				}
 			}
-		},
+		},]]--
 		--#INSERTION_MARKER# DO NOT REMOVE THIS. USED BY appscaffold SCRIPT.
 	}
 }
