Index: cmake_find_modules/Find_lua.cmake
===================================================================
--- a/cmake_find_modules/Find_lua.cmake
+++ b/cmake_find_modules/Find_lua.cmake
@@ -3,6 +3,7 @@
 	NAMES
 		lualib.h
 	PATHS
+		/usr/include/lua5.1
 		/usr/include
 		/usr/local/include
 		/usr/local/include/lua51
@@ -12,7 +13,7 @@
 
 FIND_LIBRARY(LUA_LIBRARY_PATH_
 	NAMES
-		lua
+		lua5.1
 	PATHS
 		/usr/lib64
 		/usr/lib
@@ -22,7 +23,7 @@
 		/usr/local/lib/lua51
 		/sw/lib
 		/opt/local/lib
-		NO_DEFAULT_PATH)
+		)
 
 IF(LUA_INCLUDE_PATH_)
 	SET(LUA_FOUND 1)
Index: cmake_find_modules/Find_openssl.cmake
===================================================================
--- a/cmake_find_modules/Find_openssl.cmake
+++ b/cmake_find_modules/Find_openssl.cmake
@@ -8,7 +8,7 @@
 		/usr/include
 		/usr/local/include
 		/usr/local/ssl/include
-		NO_DEFAULT_PATH)
+		)
 
 IF($ENV{COMPILE_STATIC} MATCHES "1")
 	SET(CMAKE_FIND_LIBRARY_SUFFIXES_OLD ${CMAKE_FIND_LIBRARY_SUFFIXES})
@@ -36,7 +36,7 @@
 		/lib/i386-linux-gnu
 		/usr/lib/i386-linux-gnu
 		/usr/local/ssl/lib
-		NO_DEFAULT_PATH)
+		)
 
 FIND_LIBRARY(CRYPTO_LIBRARY_PATH
 	NAMES
@@ -58,7 +58,7 @@
 		/lib/i386-linux-gnu
 		/usr/lib/i386-linux-gnu
 		/usr/local/ssl/lib
-		NO_DEFAULT_PATH)
+		)
 
 IF($ENV{COMPILE_STATIC} MATCHES "1")
 	FIND_LIBRARY(Z_LIBRARY_PATH
@@ -80,7 +80,7 @@
 			/lib/i386-linux-gnu
 			/usr/lib/i386-linux-gnu
 			/usr/local/ssl/lib
-			NO_DEFAULT_PATH)
+			)
 	SET(CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES_OLD})
 ELSEIF($ENV{COMPILE_STATIC} MATCHES "1")
 	SET(Z_LIBRARY_PATH "")
